Vision

jaimeyu opened the pull request:
I would like to merge my vision branch to the main branch. The vision branch is good shape and we should start using it for tweaking.

jaimeyu committed:
Added profiler code to see how much time operations take.
Added a counter so we can fire operations every ~1 second.
Added basic vision code and put it into a thread that fires every 1
second.

jaimeyu committed:
Added code in an attempt to reverse the rotation rate of the joystick so
left and right are now rotating the robot in the correct direction.

jaimeyu committed:
Updated the lift motor values to the tweaked one for the new motor. The
lift stop command is at 5% because without it, the lift falls down. 

Found mistake in button layout. There were two lift up when one them
should have been down.

jaimeyu committed:
Reformatted the code. 
Made the lift power to 100%.

jaimeyu committed:
Upped the lift motor power when going down
Made the arcade drive code simpler and tested it. Left is left and right
is right now.

jaimeyu committed:
Refactored the vision code into its own class. 
This cleans up the code nicely.

BrendaTruongg commented:
How do I request a pull?

On Fri, Feb 6, 2015 at 9:11 PM, Jaime Yu notifications@github.com wrote:

> I would like to merge my vision branch to the main branch. The vision
> 
> ## branch is good shape and we should start using it for tweaking.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/FRC-5652/FRC-broken-window/pull/2
> Commit Summary
> - Added profiler code to see how much time operations take.
> - Added code in an attempt to reverse the rotation rate of the
>   joystick so
> - Updated the lift motor values to the tweaked one for the new motor.
>   The
> - Reformatted the code.
> - Upped the lift motor power when going down
> - Refactored the vision code into its own class.
> 
> File Changes
> - _M_ Robotics/src/org/usfirst/frc/team5652/robot/Robot.java
>   https://github.com/FRC-5652/FRC-broken-window/pull/2/files#diff-0
>   (282)
> - _A_ Robotics/src/org/usfirst/frc/team5652/robot/Vision.java
>   https://github.com/FRC-5652/FRC-broken-window/pull/2/files#diff-1
>   (67)
> 
> Patch Links:
> - https://github.com/FRC-5652/FRC-broken-window/pull/2.patch
> - https://github.com/FRC-5652/FRC-broken-window/pull/2.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/FRC-5652/FRC-broken-window/pull/2.

## 

_ocsb.ca 
http://ocsb.ca/?utm_source=ocsbstudents&utm_medium=email_footer&utm_campaign=awareness *
| *Inspired education. Inspiring students._ 

Think "green," please consider the environment before printing this email.

*Confidentiality and privacy notice: This email, including any attachments, 
is for the sole use of the intended recipient(s) and may contain 
confidential and privileged information. Any unauthorized review, use, 
disclosure, or distribution is prohibited. If you received this email 
and are not the intended recipient, please inform the sender by email reply 
and destroy all copies of the original message. This email and any 
emails sent to or from the Ottawa Catholic School board may be 
transmitted or stored outside of Canada. *

BrendaTruongg committed:
Added motor 7
Changed motor's names from "lift_system" to "motor_x"

jaimeyu committed:
Deactivated vision code (problems with it crashing!)
Added sensitivity multiplier. When button 12 is _HELD!_, all controls
will only run at 30% of their max speed. 
Renamed all the b{n} buttons variables to what they really are.

jaimeyu committed:
Added general comments.
Changed vision code to always run in a separate thread (not relying on
teleop/auto). 
Changed vision code to lower quality, focus on red and grays, and
lowered the resolution.

jaimeyu committed:
Realized, we don't need image processing for teleop. So I put in a
boolean to control whether we want just a simple camera so the driver
can see what they're doing, or a more advance one for image processing. 

This should give us the flexibility to go to plan B (no image
processing) very easily. As well, this might be faster than manually
sending a frame (the library code says it takes 50ms to manually
transfer an image but doesn't say what the penalty is for using the
automatic capture mode, I'm going to assume its faster).

UNFORTUNATELY!!! We can't disable the simple cam server!!! The NI class
uses a thread which is not accessible outside the object. So once we
start a simple cam server, we cannot stop it. If I'm wrong, please add
the code in to allow us to switch on the fly between manual and
automatic capture.

jaimeyu committed:
Added smartdashboard string to tell the driver station which mode the
camera is in.

jaimeyu committed:
Added compressor object.
Changed solenoid code to take in a cam module number.
Added a bunch of smartdashboard diagnostics.

jaimeyu committed:
Added PDP code for diagnostics (please add diagnostics)

jaimeyu committed:
Simplified autonomous code into clear methods. Should be easier to
tweak.

BrendaTruongg committed:
- added limit switches for forklifts

BrendaTruongg committed:
- added motor 8

BrendaTruongg committed:
Tweaked the sensitivity
Autonomous tweak.

BrendaTruongg committed:
Sensitivity setting needs to be rewritten. The global is not right. 
Changed the turning sensitivity to half the speed because it was turning
too fast.

jaimeyu committed:
Added the pneumatic code back.
Added private identifier to variables.

jaimeyu committed:
Added smartdashboard put/get code.

jaimeyu committed:
Added comments for joystick

jaimeyu committed:
Added new limit switch class for digital inputs.
The reasoning is because the there is no debounce code. Check
http://whatis.techtarget.com/definition/debouncing

jaimeyu committed:
Refactored the code so its in smaller manageable chunks.

jaimeyu committed:
Tween logic prototype. 
When you hold the lift triggers for 2 seconds, the amount of power is
slowly ramped up. This way, it gives the operator a little bit of extra
sensitivity.

jaimeyu committed:
Small mistake with forklift down. last_lift_state should be DOWN not up.

jaimeyu committed:
Added to forklift methods the stop code at the end of a operation. If
the sensor is triggered at the end, then we disable the lift motors.

jaimeyu committed:
Re-ordered the tween logic in the lift code. This should be more
stateful.

jaimeyu committed:
Added comments to how the tween code works.
Refactored the tween code so it's portable for the project.

jaimeyu commented:
Too old. Issue pull on latest.
